test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ActionOverloadingTests/Tests/ActionOverloadingEndToEndTests.cs

```diff
@@ -0,0 +1,266 @@
+            IEnumerable<OperationDescriptor> specialEmployeeDescriptors = _context.GetEntityDescriptor(specialEmployee).OperationDescriptors.Where(od => od.Title.Equals(actionTitle));
+            List<Tuple<string, string>> expectedLinkValuesList = new List<Tuple<string, string>>()
+            {
+                new Tuple<string, string>(metadataPrefix + actionName, "People(-7)/Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.Employee/" + actionTitle),
+                new Tuple<string, string>(metadataPrefix + actionName, "People(-7)/Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.SpecialEmployee/" + actionTitle),
+            };
+
+            this.VerifyLinks(expectedLinkValuesList, specialEmployeeDescriptors);
+
+            _context.Execute<bool>(specialEmployeeDescriptors.Where(d => d.Target.AbsoluteUri.Contains(".Employee")).Single().Target, "POST", true, new BodyOperationParameter("n", 123)).Single();
+            _context.Execute<int>(specialEmployeeDescriptors.Where(d => d.Target.AbsoluteUri.Contains(".SpecialEmployee")).Single().Target, "POST", true).Single();
+        }
+
+        [Fact, TestPriority(2)]
+        public void ExecuteEntitySetBoundOverloadedOperations()
+        {
+            for (int i = 1; i < 2; i++)
+            {
+                // action bound with collection of employees
+                _context.Execute(
+                    new Uri(_baseUri + "People/Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.Employee/Default.IncreaseSalaries", UriKind.Absolute),
+                    "POST",
+                    new BodyOperationParameter("n", 3));
+
+                // action bound with collection of special employees
+                _context.Execute(
+                    new Uri(_baseUri + "People/Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.SpecialEmployee/Default.IncreaseSalaries", UriKind.Absolute),
+                    "POST",
+                    new BodyOperationParameter("n", 3));
+            }
+        }
+
+        [Fact]
+        public void OverloadedActionsProjection()
+        {
+            for (int i = 1; i < 2; i++)
+            {
+                Common.Clients.EndToEnd.Product product = _context.Execute<Common.Clients.EndToEnd.Product>(new Uri(_baseUri + "Products(-10)?$select=Default.*", UriKind.Absolute)).Single();
+                OperationDescriptor productOperationDescriptor = _context.GetEntityDescriptor(product).OperationDescriptors.Single();
+                Assert.Equal(_baseUri + "$metadata#Default.RetrieveProduct", productOperationDescriptor.Metadata.AbsoluteUri, true);
+                Assert.Equal(_baseUri + "Products(-10)/Default.RetrieveProduct", productOperationDescriptor.Target.AbsoluteUri, true);
+
+                Common.Clients.EndToEnd.OrderLine orderLine = _context.Execute<Common.Clients.EndToEnd.OrderLine>(new Uri(_baseUri + "OrderLines(OrderId=-10,ProductId=-10)?$select=*")).Single();

```

Comment:
Maybe you could have scenarios where you select something other than `*`, like select specific property names?